internal/release/services/release/release.go

```diff
@@ -54,6 +73,188 @@ func (s *ReleaseService) RegisterWithServer(grpcServer grpc.ServiceRegistrar) er
+
+	if request.MaxAgeSeconds == nil {
+		request.MaxAgeSeconds = &DefaultMaxAgeSeconds
+	}
+
+	// Generate UTC string (Now - MaxAgeSeconds)
+	splitTimeISO := time.Unix(time.Now().UTC().Unix()-int64(*request.MaxAgeSeconds), 0).UTC().String()
+
+	// Construct the filter builder with a name and value.
+	filter := expression.Name("createdAt").GreaterThanEqual(expression.Value(splitTimeISO))
+	expr, err := expression.NewBuilder().WithFilter(filter).Build()
+	if err != nil {
+		span.RecordError(err)
+		return nil, err
+	}
+
+	scanQueryInputs := &dynamodb.ScanInput{
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		FilterExpression:          expr.Filter(),
+		TableName:                 aws.String(ReleaseTableName),
+	}
+
+	response, err := s.Db.Scan(ctx, scanQueryInputs)
+	if err != nil {
+		span.RecordError(err)
+		return nil, err
+	}
+	span.SetAttributes(
+		attribute.Int("release.count", len(response.Items)),
+	)
+	grpcResponse := releaseSvc.ListReleasesResponse{}
+	if response.Items != nil {
+		for _, item := range response.Items {
+
+			Release := &releaseSvc.Release{}
+			if err3 := attributevalue.UnmarshalMap(item, &Release); err3 != nil {
+				span.RecordError(err3)
+				return nil, err3
+			}
+			grpcResponse.Releases = append(grpcResponse.Releases, Release)
+		}
+	}
+
+	// Sort list of releases based on createdAt field
+	grpcResponse.Releases = sortReleaseList(grpcResponse.Releases)
+
+	return &grpcResponse, nil
+}
+
+// Sort a slice of releaseSvc.Release struct, using sort.SliceStable method
+func sortReleaseList(releases []*releaseSvc.Release) []*releaseSvc.Release {
+	sort.SliceStable(releases, func(a, b int) bool {
+
+		TimeA, err := time.Parse(TimeFormatLayout, releases[a].CreatedAt)
+		if err != nil {
+			log.Printf("Failed to parse createdAt field: %v", err)
+		}
+
+		TimeB, err := time.Parse(TimeFormatLayout, releases[b].CreatedAt)
+		if err != nil {
+			log.Printf("Failed to parse createdAt field: %v", err)
+		}
+		return TimeA.Before(TimeB)
+	})
+
+	return releases
+}
+
+// TODO: OPTIMIZE!!!
+// GetLatestRelease retrieves the latest release.
+func (s *ReleaseService) GetLatestRelease(ctx context.Context, request *releaseSvc.ListReleasesRequest) (*releaseSvc.ListReleasesResponse, error) {
+
+	span := trace.SpanFromContext(ctx)
+
+	// attribute does not support Uint64 so converting to int64
+	MaxAgeSeconds := convertUint64ToInt64(request.GetMaxAgeSeconds())
+
+	span.SetAttributes(
+		attribute.StringSlice("release.organizations", request.GetOrganizations()),
+		attribute.Int64("release.maxAge", MaxAgeSeconds),
+		attribute.StringSlice("release.types", request.GetTypes()),
+		attribute.String("release.page", request.GetPage().String()),
+	)
+	defer span.End()
+	scanQueryInputs := &dynamodb.ScanInput{
+		TableName: aws.String(ReleaseTableName),
+	}
+
+	response, err := s.Db.Scan(ctx, scanQueryInputs)
+	if err != nil {
+		span.RecordError(err)
+		return nil, err
+	}
+	log.Println(response)
```

Comment:
Missed this one on my last review - remove as I don't believe it's adding anything....